modules/tribes/client/components/CirclePage.component.js

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import * as api from '../api/tribes.api';
+import JoinButton from './JoinButton';
+import { getRouteParams } from '@/modules/core/client/services/angular-compat';
+
+export default function CirclePage({ user, goBack }) {
+  const [circle, setCircle] = useState(null);
+  const { t } = useTranslation('tribes');
+
+  const fetchData = async () => {
+    const circle = await api.get(getRouteParams().circle);
+    setCircle(circle);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  // Circle loading
+  if (!circle) {
+    return (
+      <div className="container container-spacer text-muted text-center">
+        {/* <!-- tr-spinner will have `aria-busy` etc --> */}
+        {/* <tr-spinner size="sm"></tr-spinner> */}
+        <br />
+        <br />
+        <small aria-hidden="true">{t('Wait a moment...')}</small>
+      </div>
+    );
+  }
+
+  // Circle not found
+  if (!circle._id) {
```

Comment:
Might be good to protect this a bit with: `!circle?._id`; that way this check will fail if circle isn't object as well when _id doesn't exist.